### ISSUE TITLE
GetEnterpriseUsersWithMarkerAsync throws NRE when evaluating Entries

### ISSUE THREAD

**Issue by watfordsuzy** [box/box-windows-sdk-v2#948]
- [X] I have checked that the [SDK documentation][sdk-docs] doesn't solve my issue.
- [X] I have checked that the [API documentation][api-docs] doesn't solve my issue.
- [X] I have searched the [Box Developer Forums][dev-forums] and my issue isn't already reported (or if it has been reported, I have attached a link to it, for reference).
- [X] I have searched [Issues in this repo][github-repo] and my issue isn't already reported.

### Description of the Issue
If I include an `externalAppUserId` to a call to `GetEnterpriseUsersWithMarkerAsync` the resulting marker based collection has a null `Entries` field.

### Steps to Reproduce
```csharp
string? marker = null;
do
{
    var result = await client.UsersManager.GetEnterpriseUsersWithMarkerAsync(marker, externalAppUserId: "user_id");
    foreach (BoxUser user in result.Entries) // NullReferenceException (result.Entries)
    {
        // ...
    }

    marker = result.NextMarker;
}
while (!string.IsNullOrWhiteSpace(marker));
```

### Expected Behavior
The list of entries is enumerated.

### Error Message, Including Stack Trace
This is in a durable function app and the stack trace is a bit useless.

### Versions Used
.Net SDK: 5.7.0
Windows: dotnet-isolated|8.0

[sdk-docs]: ./docs
[api-docs]: https://developer.box.com/docs
[dev-forums]: https://community.box.com/t5/Platform-and-Development-Forum/bd-p/DeveloperForum
[github-repo]: https://github.com/box/box-windows-sdk-v2/search?type=Issues

**Comment by watfordsuzy** [comment]
Worse still, calling `GetEnterpriseUsersAsync` also throws an error when I include the external app ID.

**Comment by watfordsuzy** [comment]
I now think this is a regression with the main Box API, how do I get this escalated?

https://gist.github.com/watfordsuzy/c5f85f1099e259f64e5acb4b0881d18f

**Comment by mwwoda** [comment]
You also mentioned regression? Could you elaborate on that? Was it working fine for you in the past?

**Comment by watfordsuzy** [comment]
Yes, all of this code was working for me previously. Seems only recently that the addition of `external_app_user_id` causes the weird response from the server.

**Comment by watfordsuzy** [comment]
It looks like you were correct and I can successfully work around the issue using `GetEnterpriseUsersAsync`. Not 100% sure why I saw the NRE, might have been an old instance of the code was picked up by the function app host.

**Comment by mwwoda** [comment]
Service owners confirmed the issue and they looking into a fix. I don't have any timeframe for now, but I'll let you know in case of any updates. For now, my advice is to stick with `GetEnterpriseUsersAsync` function.

**Comment by watfordsuzy** [comment]
@mwwoda thank you, will do!

Is it even possible to have two box users with the same `external_app_user_id`?

**Comment by congminh1254** [comment]
Hi @watfordsuzy 

As I tested, so it's not possible to have two Box users with the same `external_app_user_id`.

**Comment by watfordsuzy** [comment]
@mwwoda @congminh1254 I received the following from Box Support on my case and I'm not sure how to manage this with the C# library:
> Case: #3099758
> Hi Christopher,
>
> Thanks for reaching out. My name is Dan and I'm a Platform specialist here at Box. I understand then when using the .NET SDK, that when you pass a external_app_user_id parameter along with a usermarker=true parameter to a List Enterprise Users call that marker-based pagination is not being used.
>
> When using marker-based pagination, a marker field will only be returned in the response when there are enough entries to warrant multiple pages of results. Where an external_app_user_id is specified only a single entry will ever be returned, and so a marker field will not be included in the results even when marker-based pagination is specified. 
>
> I hope this is helpful. Please let me know if you have any questions or if I may be of any further assistance. 

I asked a follow-up question about the stability of the APIs when requesting Marker Based results and received the following:
> Hi Christopher,
>
> Inasmuch as it's not always possible to know whether enough entries will be returned to allow for multiple page of results, it will not always be definitively possible to know whether results will be returned in a marker-based format. In this connection, the best way forward will be to write the program in such a way as to be able to useful parse results in either format.
>
> I can definitely see how it would be useful for the API to always return results in the same format when marker based pagination is specified, even in the event that only a single page of results is returned. You can flag the need for this change at [pulse.box.com](http://pulse.box.com/). Our engineers regularly review this platform to inform their future priorities and roadmaps.

**Comment by watfordsuzy** [comment]
@mwwoda @congminh1254 thoughts on this? I think the SDK should be modified such that this doesn't break unexpectedly. Either by:
1. Not allowing the external app ID when you use marker based pagination
2. "Fixing" the response from the API to match the contract of the marker parameter (very bizarre to me the API gets to ignore that field in certain situations).

**Comment by mwwoda** [comment]
@watfordsuzy 
I just ping the team to check the state of the fix on the backend. Ideally, using externalAppId should not lead to this behaviour, and SDK fix should not be necessary. If it cannot be fixed on the service side for some reason, then we need to adjust the SDK.
